File: whdon_app/src/main/java/nl/whitedove/washetdroogofniet/CustomListAdapterTotStats.java
package nl.whitedove.washetdroogofniet;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.Filter;
import android.widget.Filterable;
import android.widget.ProgressBar;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

class CustomListAdapterTotStats extends BaseAdapter implements Filterable {

    private List<Statistiek> listData;
    private List<Statistiek>filteredData = null;
    private ItemFilter mFilter = new ItemFilter();

    private LayoutInflater layoutInflater;

    CustomListAdapterTotStats(Context context, ArrayList<Statistiek> listData) {

        this.listData = listData;
        this.filteredData = listData ;
        layoutInflater = LayoutInflater.from(context);
    }

    @Override
    public int getCount() {
        return filteredData.size();
    }

    @Override
    public Object getItem(int position) {
        return filteredData.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @SuppressLint("DefaultLocale")
    public View getView(int position, View convertView, ViewGroup parent) {

        ViewHolder holder;
        if (convertView == null) {
            convertView = layoutInflater.inflate(R.layout.totstat_list_layout, parent, false);
            holder = new ViewHolder();
            holder.tvLocatie = convertView.findViewById(R.id.tvLocatie);
            holder.tvLocatieAantal = convertView.findViewById(R.id.tvLocatieAantal);
            holder.tvDroog = convertView.findViewById(R.id.tvDroog);
            holder.tvNat = convertView.findViewById(R.id.tvNat);
            holder.pbProgress = convertView.findViewById(R.id.pbDroogNatStat);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }

        Statistiek stat = filteredData.get(position);
        int aantalDroog = stat.getAantalDroog();
        int aantalNat = stat.getAantalNat();
        int totaal = aantalDroog + aantalNat;

        holder.tvLocatie.setText(stat.getLocatie());
        holder.tvLocatieAantal.setText(String.format("(%d)", totaal));
        holder.pbProgress.setProgress(100 * aantalDroog / totaal);
        int percDroog = Math.round(100.0F * aantalDroog / totaal);
        int percNat = 100 - percDroog;

        holder.tvDroog.setText(String.format("%d%%", percDroog));
        holder.tvNat.setText(String.format("%d%%", percNat));
        return convertView;
    }

    private static class ViewHolder {
        TextView tvLocatie;
        TextView tvLocatieAantal;
        TextView tvDroog;
        TextView tvNat;
        ProgressBar pbProgress;
    }

    public Filter getFilter() {
        return mFilter;
    }

    private class ItemFilter extends Filter {
        @Override
        protected FilterResults performFiltering(CharSequence constraint) {

            String filterString = constraint.toString().toLowerCase();

            FilterResults results = new FilterResults();

            final List<Statistiek> list = listData;

            int count = list.size();
            final ArrayList<Statistiek> nlist = new ArrayList<>(count);

            Statistiek filterableStat ;

            for (int i = 0; i < count; i++) {
                filterableStat = list.get(i);
                if (filterableStat.getLocatie().toLowerCase().contains(filterString)) {
                    nlist.add(filterableStat);
                }
            }

            results.values = nlist;
            results.count = nlist.size();

            return results;
        }

        @SuppressWarnings("unchecked")
        @Override
        protected void publishResults(CharSequence constraint, FilterResults results) {
            filteredData = (ArrayList<Statistiek>) results.values;
            notifyDataSetChanged();
        }
    }
}

File: whdon_app/src/main/java/nl/whitedove/washetdroogofniet/CustomListAdapterTotStats.kt
package nl.whitedove.washetdroogofniet

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.Filter
import android.widget.Filterable
import android.widget.ProgressBar
import android.widget.TextView

import java.util.ArrayList

internal class CustomListAdapterTotStats(context: Context, listData: ArrayList<Statistiek>) : BaseAdapter(), Filterable {

    private val listData: List<Statistiek>
    private var filteredData: List<Statistiek>? = null
    private val mFilter = ItemFilter()

    private val layoutInflater: LayoutInflater

    init {

        this.listData = listData
        this.filteredData = listData
        layoutInflater = LayoutInflater.from(context)
    }

    override fun getCount(): Int {
        return filteredData!!.size
    }

    override fun getItem(position: Int): Any {
        return filteredData!![position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    @SuppressLint("DefaultLocale")
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var cv = convertView
        val holder: ViewHolder
        if (cv == null) {
            cv = layoutInflater.inflate(R.layout.totstat_list_layout, parent, false)
            holder = ViewHolder()
            holder.tvLocatie = cv!!.findViewById(R.id.tvLocatie)
            holder.tvLocatieAantal = cv.findViewById(R.id.tvLocatieAantal)
            holder.tvDroog = cv.findViewById(R.id.tvDroog)
            holder.tvNat = cv.findViewById(R.id.tvNat)
            holder.pbProgress = cv.findViewById(R.id.pbDroogNatStat)
            cv.tag = holder
        } else {
            holder = cv.tag as ViewHolder
        }

        val stat = filteredData!![position]
        val aantalDroog = stat.aantalDroog
        val aantalNat = stat.aantalNat
        val totaal = aantalDroog + aantalNat

        holder.tvLocatie!!.text = stat.locatie
        holder.tvLocatieAantal!!.text = String.format("(%d)", totaal)
        holder.pbProgress!!.progress = 100 * aantalDroog / totaal
        val percDroog = Math.round(100.0f * aantalDroog / totaal)
        val percNat = 100 - percDroog

        holder.tvDroog!!.text = String.format("%d%%", percDroog)
        holder.tvNat!!.text = String.format("%d%%", percNat)
        return cv
    }

    private class ViewHolder {
        internal var tvLocatie: TextView? = null
        internal var tvLocatieAantal: TextView? = null
        internal var tvDroog: TextView? = null
        internal var tvNat: TextView? = null
        internal var pbProgress: ProgressBar? = null
    }

    override fun getFilter(): Filter {
        return mFilter
    }

    private inner class ItemFilter : Filter() {
        override fun performFiltering(constraint: CharSequence): Filter.FilterResults {

            val filterString = constraint.toString().toLowerCase()

            val results = Filter.FilterResults()

            val list = listData

            val count = list.size
            val nlist = ArrayList<Statistiek>(count)

            var filterableStat: Statistiek

            for (i in 0 until count) {
                filterableStat = list[i]
                if (filterableStat.locatie!!.toLowerCase().contains(filterString)) {
                    nlist.add(filterableStat)
                }
            }

            results.values = nlist
            results.count = nlist.size

            return results
        }

        override fun publishResults(constraint: CharSequence, results: Filter.FilterResults) {
            @Suppress("UNCHECKED_CAST")
            filteredData = results.values as ArrayList<Statistiek>
            notifyDataSetChanged()
        }
    }
}

